src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpTraceHelper.cs

```diff
@@ -33,7 +33,17 @@ public static bool IsTraceEnabled()
             return s_TraceEnabled;
         }
 
-        public static void TraceCallbackStatus(string message, IntPtr handle, bool invokeCallback, uint status)
+        public static void Trace(string format, params object[] args)
```

Comment:
If you pass any args to Trace, the compiler is going to allocate an array to hold them even if !IsTraceEnabled.  For now you could add a ```[Conditional("DEBUG")]``` to these Trace methods, as they're just calling Debug.WriteLine if enabled and so they're only useful if DEBUG is set.  Longer term, I know you're planning on converting the tracing to using an EventSource, at which point you'd want some other solution to ensure that such allocations don't happen if tracing isn't enabled.